client/galaxy/scripts/layout/menu.js

```diff
@@ -222,25 +222,38 @@ var Collection = Backbone.Collection.extend({
         // User tab.
         //
         if ( !Galaxy.user.id ){
-            var userTab = {
-                id              : 'user',
-                title           : _l('Login or Register'),
-                cls             : 'loggedout-only',
-                tooltip         : _l('Account registration or login'),
-                menu            : [{
-                    title           : _l('Login'),
-                    url             : 'user/login',
-                    target          : 'galaxy_main',
-                    noscratchbook   : true
-                }]
-            };
-            options.allow_user_creation && userTab.menu.push({
-                title           : _l('Register'),
-                url             : 'user/create',
-                target          : 'galaxy_main',
-                noscratchbook   : true
-            });
-            this.add( userTab );
+            if ( options.allow_user_creation ) {
+                var userTab = {
+                    id              : 'user',
+                    title           : _l('Login or Register'),
+                    cls             : 'loggedout-only',
+                    tooltip         : _l('Account registration or login'),
+                    menu            : [{
+                            title           : _l('Login'),
+                            url             : 'user/login',
+                            target          : 'galaxy_main',
+                            noscratchbook   : true
+                        }, {
+                            title: _l('Register'),
+                            url: 'user/create',
+                            target: 'galaxy_main',
+                            noscratchbook: true
+                        }
+                    ]
+                };
+                this.add(userTab);
```

Comment:
This line is present in both the `if` and `else` branch, I think you can move it after the end of the `else` branch, as it was before.